README.md

```diff
@@ -55,25 +61,64 @@ spec:
   restartPolicy: Always
   ```
 
-## Installing a different Node.js version
+### Installing a Different Node.js Version
 
 The image includes the `install_node` script that can be used to install a particular Node.js version. For example:
 
-        FROM launcher.gcr.io/google/nodejs
+```docker
+FROM launcher.gcr.io/google/nodejs
 
-        # Install node.js 4.6.1
-        RUN install_node v4.6.1
+# Install node.js 4.6.1
+RUN install_node v4.6.1
 
-        # Copy application code.
-        COPY . /app/
+# Copy application code.
+COPY . /app/
 
-        # Install dependencies.
-        RUN npm --unsafe-perm install
+# Install dependencies.
+RUN npm --unsafe-perm install
+```
 
 Node.js is installed with binary packages hosted on a Google-provided mirror.
 
+## The Node.js Runtime Builder
 
-# Contributing changes
+The Node.js Runtime Builder is used to deploy applications to App Engine Flexible when the `gcloud beta app deploy` command is used.  To deploy your application, issue `gcloud beta app deploy` from your application's root directory containing its `package.json` file.
+
+During the deployment process, the Runtime Builder analyzes the files in your application's root directory to determine how to deploy your application.  The specifics with respect to what is used in the deployment is outlined in the following sections.  Additional information can be found in the Google Cloud Platform Node.js Runtime [documentation](https://cloud.google.com/appengine/docs/flexible/nodejs/runtime).
+
+### Node.js Version
+
+If your application's `package.json` file contains an `engines` field that contains a `node` field, that field will be used as a semver specification for the Node.js version to use in the deployment.  The Node.js version that matches that semver string will be used to run your deployed application.  If no Node.js version matches the semver string specified, the deployment will be aborted.
+
+The following illustrates a section of a `package.json` file that specifies that the latest version of Node.js 6 should be used:
+```javascript
+{
+  ...
+  "engines": {
+    "node": "6.x.x"
+  }
+  ...
+}
+```
+
+### Package Manager
+
+If your application's root directory contains a `yarn.lock` file, and `yarn.lock` is not specified in the `skip_files` section of your application's `app.yaml` file, then `yarn` will be used to install your application's dependencies and start the application.
+
+Otherwise, if a `yarn.lock` file does not exist in your application's root directory, or if `yarn.lock` is specified in the `skip_files` section of your application's `app.yaml` file, `npm` will be used to install dependencies and start your application.
```

Comment:
No need to restate the negative of the previous paragraph. "Otherwise, `npm` will be used to install dependencies and start your application"